madmom/ml/hmm.pyx

```diff
@@ -537,12 +584,8 @@ class HiddenMarkovModel(object):
         cdef double prob_sum, norm_factor
 
         # init forward variables
-        if init is None:
-            for state in range(num_states):
-                fwd[0, state] = self.initial_distribution[state]
-        else:
-            for state in range(num_states):
-                fwd[0, state] = init[state]
+        for state in range(self.initial_distribution.shape[0]):
+            fwd[0, state] = self.initial_distribution[state]
```

Comment:
Please revert these changes, since this change is required by other code to work.

NB: it is likely that the ``init`` parameter will be removed later on, since it is a very dirty way of keeping the state of the HMM, but right now it is required.